### PR TITLE
Warn about literal references to Emacs user directory or files.

### LINKED ISSUE #24: Warn about literal references to ~/.emacs.d
A library should use `user-emacs-directory` for the path to the Emacs directory, and `locate-user-emacs-file` to find its data or config files within this directory.

I'll open a PR if I can find some time.

### PR DESCRIPTION
**Pull request by alphapapa**
Fixes #24.

### CODE REVIEW

**Comment by alphapapa** [comment]
If I'm reading the logs correctly, it's failing because it's detecting itself...  LOL  :)

File: package-lint.el [hunk 1]
```diff
@@ -253,6 +254,15 @@ This is bound dynamically while the checks run.")
 
 ;;; Checks
 
+(defun package-lint--check-for-literal-emacs-path ()
+  "Verify that package does not refer to \"~/.emacs.d\" literally.
+Instead it should use `user-emacs-directory' and `locate-user-emacs-file'."
+  (cl-loop
+   while (re-search-forward (rx (syntax string-quote) (1+ (not (syntax string-quote))) (or "/" "\\") ".emacs.d") nil t)
```
Comment (Fanael) [inline package-lint.el]: Why `cl-loop` instead of just `while`?

Comment (alphapapa) [inline package-lint.el]: Ah, good point, I whittled down the contents of the loop but didn't think to change it to `while`.

**Comment by purcell** [comment]
> If I'm reading the logs correctly, it's failing because it's detecting itself... LOL :)

LOL indeed! You could cheat by using something like `\.emacs\.d`?

**Comment by alphapapa** [comment]
Thanks, I think that should do it.  :)

**Comment by Fanael** [comment]
Tests are missing.

**Comment by alphapapa** [comment]
Guess I haven't been paying attention, I didn't think about that.  Looks simple, I will do that in the next day or two.  Thanks.

**Comment by alphapapa** [comment]
Test added, seems to work.  Hope I did everything right.  :)

**Comment by Fanael** [comment]
I amended it to fix the lint failure, merged as f41280e1b.

**Comment by alphapapa** [comment]
@Fanael Sorry if I'm being dense, but as far as I can tell, https://github.com/purcell/package-lint/commit/f41280e1bc67a78eef0a3c28f8a73dc4521a5875 is exactly the same as https://github.com/purcell/package-lint/pull/58/commits/f8dcb552e52fa6e4538c791502ec64ee37421509, which is failing the CI, so the CI should fail with https://github.com/purcell/package-lint/commit/f41280e1bc67a78eef0a3c28f8a73dc4521a5875 also, right?

**Pull request closed without merging**